lib/matplotlib/backends/backend_agg.py

```diff
@@ -485,6 +485,44 @@ def print_raw(self, filename_or_obj, *args, **kwargs):
     print_rgba = print_raw
 
     def print_png(self, filename_or_obj, *args, **kwargs):
+        """
+        Write the png data to the given file.
+
+        Parameters
+        ----------
+        filename_or_obj : str or PathLike or file-like object
+            The file to write to.
+
+        metadata : dict, optional
+            Metadata in the PNG file as key-value pairs of bytes or latin-1
+            encodable strings.
+            According to the PNG specification, keys must be shorter than 79
+            chars.
+
+            The PNG specification defines some common keywords that may be
+            used as appropriate:
+            - Title: Short (one line) title or caption for image.
+            - Author: Name of image's creator.
+            - Description: Description of image (possibly long).
+            - Copyright: Copyright notice.
+            - Creation Time: Time of original image creation
+              (usually RFC 1123 format).
+            - Software: Software used to create the image.
+            - Disclaimer: Legal disclaimer.
+            - Warning: Warning of nature of content.
+            - Source: Device used to create the image.
+            - Comment: Miscellaneous comment;
+              conversion from other image format.
+
+            Other keywords may be invented for other purposes.
+
+            If 'Software' is not given, an autogenerated value for matplotlib
+            will be used.
+
+            For more details see the PNG specification:
+            https://www.w3.org/TR/2003/REC-PNG-20031110/#11keywords
```

Comment:
Not a huge deal but I've grown fond of out-of-line rst links, e.g. here I'd have written before the list
```
`The PNG specification`_ defines some common keywords ...
```
and at the bottom
```
_The PNG specification: https://...
```

This way the plain text is just as readable and you get just a link in the HTML, which is normally what you want.

Reply:
Does this work with line lengths? The link definition at the bottom would exceed 79 chars.

Reply:
```
_The PNG specification: \
   https://...
```
should also work (as long as the backslash stays an escape -- don't turn the docstring into a raw string!)